wts/blueprints/aggregate.py

```diff
@@ -0,0 +1,115 @@
+import asyncio
+import flask
+import httpx
+import time
+from cdiserrors import NotFoundError, UserError
+
+from ..auth import authenticate
+from ..models import db, RefreshToken
+from ..tokens import async_get_access_token
+
+
+blueprint = flask.Blueprint("aggregate", __name__)
+
+
+@blueprint.route("/<path:endpoint>", methods=["GET"])
+async def get_aggregate_response(endpoint):
+    """
+    Proxy GET requests to `endpoint` on each linked commons and return
+    an aggregated response.
+
+    For an authenticated request, each proxied request incudes an access token
+    fetched using the current user's refresh token.
+
+    The size of the aggregated response body can be reduced by supplying a
+    `filters` parameter. If provided, only return JSON key-value pairs whose
+    key is in `filters`. Multiple filters can be specified:
+
+    `GET /aggregate/user/user?filters=authz&filters=username`
+    """
+    # for `GET /aggregate/user/user`, flask sets endpoint to 'user/user'
+    endpoint = f"/{endpoint}"
+    if endpoint not in flask.current_app.config["AGGREGATE_ENDPOINT_ALLOWLIST"]:
+        raise NotFoundError(
+            "supplied endpoint is not configured in the Workspace Token Service aggregate endpoint allowlist"
+        )
+
+    filters = flask.request.args.getlist("filters")
+    parameters = flask.request.args.to_dict()
+    parameters.pop("filters", None)
+
+    if flask.request.headers.get("Authorization"):
+        authenticate(allow_access_token=True)
```

Comment:
this requires a token for the location WTS is, right?

Reply:
If so, I might suggest making this check earlier or making the `AGGREGATE_ENDPOINT_ALLOWLIST` later. We want to reduce the information someone can get from the API if they're not authorized (e.g. make the check before providing an error telling them certain endpoints aren't allowed).

Reply:
If this resource is at say `https://gen3_commons.io/wts/aggregate/authz/mapping`, the allowed issuer for the access token being validated is `https://gen3_commons.io/user`.

Reply:
If we make the aggregate endpoint such that it requires authentication no matter what, I think it makes complete sense to authenticate before checking the endpoint allowlist.

On the other hand, if we support an unauthenticated option, I'm not sure there's any way to prevent an unauthenticated user from being able to determine what the endpoint allowlist is.